docker/Dockerfile

```diff
@@ -0,0 +1,23 @@
+#Import Node base image
+FROM node:12-slim
+
+#Install Git
+RUN apt-get update && \
+    apt list -a git && \
+    apt-get install -y --no-install-recommends git=1:2.*
+
+#Clone SDK repo to app directory
+RUN git clone https://github.com/watson-developer-cloud/node-sdk.git
+
+#Install the SDK
+RUN npm install ibm-watson
```

Comment:
If we are building the SDK from source, why install it from NPM as well? Or vice versa? It seems like we would just need one or the other.

I'm no Docker expert so I guess I'm just curious to hear your reasoning!

Reply:
I build the SDK from the source in order to be able to verify the installation via the test cases. I was unsure if it was possible to just build the test cases and not install the SDK from the source and rather only install the SDK with npm as the original intention was to install the SDK as a "user" rather than a "developer".

Reply:
Hmm I see what you were going for, but I'm not sure it's necessary. Running `npm install ibm-watson` without any failures implies a successful installation - that's how we expect users to install the package in the wild. Downloading the source and running the tests just makes sure that the cloning of the repo was successful - it's independent of the package install from `npm`.

If you really wanted to double-check that the SDK is ready and usable after `npm install`, you could write a script that checked for the presence of the files in the `node_modules/` folder. Or, you could include a JS file that uses that imports the SDK but doesn't run any operations, then make sure that runs without error.